browser/ai_chat/upload_file_helper.cc

```diff
@@ -165,105 +196,149 @@ void UploadFileHelper::MultiFilesSelected(
-                    mojom::UploadedFileType::kImage));
+                    filepath.AsUTF8Unsafe(), file_data->size(), *file_data,
+                    *file_type_opt));
               }
             }
             std::move(callback).Run(
                 uploaded_files.empty()
                     ? std::nullopt
                     : std::make_optional(std::move(uploaded_files)));
           },
-          std::move(upload_image_callback_)));
+          std::move(upload_file_callback_)));
 
   // Process each file
   for (const auto& file : files) {
-    auto read_image = base::BindOnce(
+    auto read_file = base::BindOnce(
         [](const ui::SelectedFileInfo& info)
-            -> std::tuple<std::optional<std::vector<uint8_t>>, std::string> {
-          return std::make_tuple(
-              ai_chat::ReadFileToBytes(info.path()),
-              base::FilePath(info.display_name).AsUTF8Unsafe());
+            -> std::tuple<std::optional<std::vector<uint8_t>>, base::FilePath> {
+          return std::make_tuple(ai_chat::ReadFileToBytes(info.path()),
+                                 base::FilePath(info.display_name));
         },
         file);
 
-    auto on_image_read = base::BindOnce(
+    auto on_file_read = base::BindOnce(
         [](base::OnceCallback<void(
-               std::tuple<std::optional<std::vector<uint8_t>>, std::string>)>
-               callback,
-           std::tuple<std::optional<std::vector<uint8_t>>, std::string>
+               std::tuple<std::optional<std::vector<uint8_t>>, base::FilePath,
+                          std::optional<mojom::UploadedFileType>>)> callback,
+           std::tuple<std::optional<std::vector<uint8_t>>, base::FilePath>
                result) {
-          auto image_data = std::get<0>(result);
-          if (!image_data) {
-            std::move(callback).Run(
-                std::make_tuple(std::nullopt, std::get<1>(result)));
+          auto file_data = std::get<0>(result);
+          auto filepath = std::get<1>(result);
+
+          if (!file_data) {
+            std::move(callback).Run(std::make_tuple(
+                std::nullopt, std::move(filepath), std::nullopt));
             return;
           }
-          ProcessImageData(
-              std::move(*image_data),
-              base::BindOnce(
-                  [](base::OnceCallback<void(
-                         std::tuple<std::optional<std::vector<uint8_t>>,
-                                    std::string>)> callback,
-                     std::string filename,
-                     std::optional<std::vector<uint8_t>> output) {
-                    std::move(callback).Run(std::make_tuple(
-                        std::move(output), std::move(filename)));
-                  },
-                  std::move(callback), std::get<1>(result)));
+
+          // Determine file type based on extension and validate PDF content
+          auto file_type_opt =
+              DetermineFileTypeFromExtension(filepath, *file_data);
```

Comment:
nit: Does a little more than just a file type check so maybe rename to `DetermineFileType`